### ISSUE TITLE
After modifying the configuration center configuration of Nacos, errors will continue to be reported

### ISSUE THREAD

**Issue by acvount** [apache/dubbo#12706]
* Dubbo version: 3.2.3
* Operating System version: MacOS 13.4.1 (22F82)
* Java version: coretto-17

 ther is my nacos config
    nacos:
      server-addr: 127.0.0.1:8848
      username: nacos
      password: nacos
      discovery:
        namespace: a995593e-7d48-4a3b-82fd-00397d9490e7
      config:
        namespace: a995593e-7d48-4a3b-82fd-00397d9490e7
        file-extension: yml
        extension-configs:
          - data-id: common-mybatis-plus-dev.yml
            refresh: true
          - data-id:dubbo-provider-dev.yml
            refresh: true
            
dubbo config:
dubbo:
  application:
    name: keyword-dubbo-service
    metadata-type: local
    qos-enable: false

  protocol:
    name: dubbo
    port: -1

  registry:
    address: nacos://127.0.0.1:8848
    username: nacos
    password: nacos
    group: DEFAULT_GROUP
    parameters:
      "namespace": a995593e-7d48-4a3b-82fd-00397d9490e7
    check: false
    register-mode: interface
    use-as-metadata-center: false
    se-as-config-center: false

  consumer:
    check: false
    timeout: 3000
            

1. start dubbo-provider-service
2. modify nacos config (Any referenced config file)
3. dubbo-provider-service error：

### Actual Behavior

If there is an exception, please attach the exception trace:

```
org.springframework.boot.context.properties.ConfigurationPropertiesBindException: Error creating bean with name 'dubbo-org.apache.dubbo.spring.boot.autoconfigure.DubboConfigurationProperties': Could not bind properties to 'DubboConfigurationProperties' : prefix=dubbo, ignoreInvalidFields=false, ignoreUnknownFields=true
	at org.springframework.boot.context.properties.ConfigurationPropertiesBindingPostProcessor.bind(ConfigurationPropertiesBindingPostProcessor.java:99) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.ConfigurationPropertiesBindingPostProcessor.postProcessBeforeInitialization(ConfigurationPropertiesBindingPostProcessor.java:79) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyBeanPostProcessorsBeforeInitialization(AbstractAutowireCapableBeanFactory.java:419) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1762) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:410) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.cloud.context.properties.ConfigurationPropertiesRebinder.rebind(ConfigurationPropertiesRebinder.java:138) ~[spring-cloud-context-4.0.3.jar:4.0.3]
	at org.springframework.cloud.context.properties.ConfigurationPropertiesRebinder.rebind(ConfigurationPropertiesRebinder.java:97) ~[spring-cloud-context-4.0.3.jar:4.0.3]
	at org.springframework.cloud.context.properties.ConfigurationPropertiesRebinder.rebind(ConfigurationPropertiesRebinder.java:85) ~[spring-cloud-context-4.0.3.jar:4.0.3]
	at org.springframework.cloud.context.properties.ConfigurationPropertiesRebinder.onApplicationEvent(ConfigurationPropertiesRebinder.java:170) ~[spring-cloud-context-4.0.3.jar:4.0.3]
	at org.springframework.cloud.context.properties.ConfigurationPropertiesRebinder.onApplicationEvent(ConfigurationPropertiesRebinder.java:53) ~[spring-cloud-context-4.0.3.jar:4.0.3]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:172) ~[spring-context-6.0.9.jar:6.0.9]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:165) ~[spring-context-6.0.9.jar:6.0.9]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:143) ~[spring-context-6.0.9.jar:6.0.9]
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:437) ~[spring-context-6.0.9.jar:6.0.9]
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:370) ~[spring-context-6.0.9.jar:6.0.9]
	at org.springframework.cloud.context.refresh.ContextRefresher.refreshEnvironment(ContextRefresher.java:103) ~[spring-cloud-context-4.0.3.jar:4.0.3]
	at org.springframework.cloud.context.refresh.ContextRefresher.refresh(ContextRefresher.java:94) ~[spring-cloud-context-4.0.3.jar:4.0.3]
	at org.springframework.cloud.endpoint.event.RefreshEventListener.handle(RefreshEventListener.java:72) ~[spring-cloud-context-4.0.3.jar:4.0.3]
	at org.springframework.cloud.endpoint.event.RefreshEventListener.onApplicationEvent(RefreshEventListener.java:61) ~[spring-cloud-context-4.0.3.jar:4.0.3]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:172) ~[spring-context-6.0.9.jar:6.0.9]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:165) ~[spring-context-6.0.9.jar:6.0.9]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:143) ~[spring-context-6.0.9.jar:6.0.9]
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:437) ~[spring-context-6.0.9.jar:6.0.9]
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:370) ~[spring-context-6.0.9.jar:6.0.9]
	at com.alibaba.cloud.nacos.refresh.NacosContextRefresher$1.innerReceive(NacosContextRefresher.java:118) ~[spring-cloud-starter-alibaba-nacos-config-2022.0.0.0-RC2.jar:2022.0.0.0-RC2]
	at com.alibaba.nacos.api.config.listener.AbstractSharedListener.receiveConfigInfo(AbstractSharedListener.java:40) ~[nacos-client-2.2.1.jar:na]
	at com.alibaba.nacos.client.config.impl.CacheData.lambda$safeNotifyListener$1(CacheData.java:334) ~[nacos-client-2.2.1.jar:na]
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[na:na]
	at java.base/java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:264) ~[na:na]
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[na:na]
	at java.base/java.lang.Thread.run(Thread.java:833) ~[na:na]
Caused by: org.springframework.boot.context.properties.bind.BindException: Failed to bind properties under 'dubbo.consumer.exported-urls[0].or-default-framework-model.model-environment' to org.apache.dubbo.common.config.Environment
	at org.springframework.boot.context.properties.bind.Binder.handleBindError(Binder.java:387) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder.bind(Binder.java:347) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder.lambda$bindDataObject$4(Binder.java:472) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.JavaBeanBinder.bind(JavaBeanBinder.java:98) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.JavaBeanBinder.bind(JavaBeanBinder.java:86) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.JavaBeanBinder.bind(JavaBeanBinder.java:62) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder.lambda$bindDataObject$5(Binder.java:476) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder$Context.withIncreasedDepth(Binder.java:590) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder$Context.withDataObject(Binder.java:576) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder.bindDataObject(Binder.java:474) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder.bindObject(Binder.java:414) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder.bind(Binder.java:343) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder.lambda$bindDataObject$4(Binder.java:472) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.JavaBeanBinder.bind(JavaBeanBinder.java:98) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.JavaBeanBinder.bind(JavaBeanBinder.java:86) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.JavaBeanBinder.bind(JavaBeanBinder.java:62) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder.lambda$bindDataObject$5(Binder.java:476) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder$Context.withIncreasedDepth(Binder.java:590) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder$Context.withDataObject(Binder.java:576) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder.bindDataObject(Binder.java:474) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder.bindObject(Binder.java:414) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder.bind(Binder.java:343) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder.lambda$bindAggregate$1(Binder.java:435) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder$Context.withSource(Binder.java:566) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder.lambda$bindAggregate$2(Binder.java:436) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.IndexedElementsBinder.bindIndexed(IndexedElementsBinder.java:106) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.IndexedElementsBinder.bindIndexed(IndexedElementsBinder.java:86) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.IndexedElementsBinder.bindIndexed(IndexedElementsBinder.java:70) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.CollectionBinder.bindAggregate(CollectionBinder.java:49) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.AggregateBinder.bind(AggregateBinder.java:56) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder.lambda$bindAggregate$3(Binder.java:438) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder$Context.withIncreasedDepth(Binder.java:590) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder.bindAggregate(Binder.java:438) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder.bindObject(Binder.java:399) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder.bind(Binder.java:343) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder.lambda$bindDataObject$4(Binder.java:472) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.JavaBeanBinder.bind(JavaBeanBinder.java:98) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.JavaBeanBinder.bind(JavaBeanBinder.java:86) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.JavaBeanBinder.bind(JavaBeanBinder.java:62) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder.lambda$bindDataObject$5(Binder.java:476) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder$Context.withIncreasedDepth(Binder.java:590) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder$Context.withDataObject(Binder.java:576) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder.bindDataObject(Binder.java:474) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder.bindObject(Binder.java:414) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder.bind(Binder.java:343) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder.lambda$bindDataObject$4(Binder.java:472) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.JavaBeanBinder.bind(JavaBeanBinder.java:98) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.JavaBeanBinder.bind(JavaBeanBinder.java:86) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.JavaBeanBinder.bind(JavaBeanBinder.java:62) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder.lambda$bindDataObject$5(Binder.java:476) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder$Context.withIncreasedDepth(Binder.java:590) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder$Context.withDataObject(Binder.java:576) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder.bindDataObject(Binder.java:474) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder.bindObject(Binder.java:414) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder.bind(Binder.java:343) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder.bind(Binder.java:332) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder.bind(Binder.java:262) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder.bind(Binder.java:249) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.ConfigurationPropertiesBinder.bind(ConfigurationPropertiesBinder.java:93) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.ConfigurationPropertiesBindingPostProcessor.bind(ConfigurationPropertiesBindingPostProcessor.java:96) ~[spring-boot-3.1.0.jar:3.1.0]
	... 32 common frames omitted
Caused by: java.lang.IllegalStateException: Unable to get value for property model-environment
	at org.springframework.boot.context.properties.bind.JavaBeanBinder$BeanProperty.lambda$getValue$0(JavaBeanBinder.java:385) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.DefaultBindConstructorProvider.getBindConstructor(DefaultBindConstructorProvider.java:46) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.ValueObjectBinder$ValueObject.get(ValueObjectBinder.java:190) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.ValueObjectBinder.bind(ValueObjectBinder.java:67) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder.lambda$bindDataObject$5(Binder.java:476) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder$Context.withIncreasedDepth(Binder.java:590) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder$Context.withDataObject(Binder.java:576) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder.bindDataObject(Binder.java:474) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder.bindObject(Binder.java:414) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.context.properties.bind.Binder.bind(Binder.java:343) ~[spring-boot-3.1.0.jar:3.1.0]
	... 90 common frames omitted
Caused by: java.lang.reflect.InvocationTargetException: null
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:568) ~[na:na]
	at org.springframework.boot.context.properties.bind.JavaBeanBinder$BeanProperty.lambda$getValue$0(JavaBeanBinder.java:382) ~[spring-boot-3.1.0.jar:3.1.0]
	... 99 common frames omitted
Caused by: java.lang.UnsupportedOperationException: Environment is inaccessible for FrameworkModel
	at org.apache.dubbo.rpc.model.FrameworkModel.getModelEnvironment(FrameworkModel.java:371) ~[dubbo-3.2.3.jar:3.2.3]
	... 104 common frames omitted
```

**Comment by acvount** [comment]
org.apache.dubbo.rpc.model.FrameworkModel.getModelEnvironment()
  throw new UnsupportedOperationException("Environment is inaccessible for FrameworkModel")

**Comment by AlbumenJ** [comment]
Fixed in #12551

**Comment by AlbumenJ** [comment]
Please try 3.2.4
